### PR TITLE
core: check return value from tee_mm_init()

### PR DESCRIPTION
**Pull request by sahilnxp**
Check return value from tee_mm_init() function.

### CODE REVIEW

File: core/mm/phys_mem.c [hunk 1]
```diff
@@ -32,7 +32,8 @@ static tee_mm_pool_t *init_pool(paddr_t b, paddr_size_t sz, uint32_t flags)
 	if (!pool)
 		panic();
 
-	tee_mm_init(pool, b, sz, CORE_MMU_USER_CODE_SHIFT, flags);
+	if (!tee_mm_init(pool, b, sz, CORE_MMU_USER_CODE_SHIFT, flags))
+		panic("Could not create memory pool");
```
Comment (etienne-lms) [inline core/mm/phys_mem.c:36]: Prefer with an empty line below.
(by the way, i'm not sure the trace message is really needed, up to you)

**Review by etienne-lms — COMMENTED** [review]
Could you change the commit message header prefix to `core: mm:` and add
`Fixes: c596d8359eb3 ("core: add phys_mem allocation functions")` tag, for consitency.

File: core/mm/phys_mem.c (hunk 1 above)
Comment (sahilnxp) [reply core/mm/phys_mem.c]: Got it.

**Comment by sahilnxp** [comment]
@etienne-lms Thanks for quick review, PR updated.

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@foss.st.com>`

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by sahilnxp** [comment]
Thanks @etienne-lms @jenswi-linaro 
Tags updated.

**Pull request merged**